src/ape_ethereum/ecosystem.py

```diff
@@ -320,11 +325,69 @@ def create_transaction(self, **kwargs) -> TransactionAPI:
+        if not matching_logs:
+            raise DecodingError(f"No logs found with ID '{abi.selector}'.")
+
+        # Process indexed data (topics)
+        log_topics = [i for i in abi.inputs if i.indexed]
+        log_topic_names = [abi.name for abi in log_topics]
+        normalized_log_topics = [abi for abi in _normalize_abi_types(log_topics)]
+        log_topic_types = [t for t in _get_abi_types(normalized_log_topics)]
+
+        # Process the rest of the log data
+        log_data = [i for i in abi.inputs if not i.indexed]
+        log_data_names = [abi.name for abi in log_data]
+        normalized_log_data = [abi for abi in _normalize_abi_types(log_data)]
+        log_data_types = [t for t in _get_abi_types(normalized_log_data)]
+
+        # Sanity check that there are not name intersections between the topic
+        # names and the data argument names.
+        duplicate_names = set(log_topic_names).intersection(log_data_names)
+        if duplicate_names:
+            duplicate_names_str = ", ".join(duplicate_names)
+            raise DecodingError(
+                "The following argument names are duplicated "
+                f"between event inputs: '{duplicate_names_str}'."
+            )
+
+        for log in matching_logs:
+            indexed_data = log["topics"] if log.get("anonymous", False) else log["topics"][1:]
+            log_data = hexstr_if_str(to_bytes, log["data"])
+
+            if len(indexed_data) != len(log_topic_types):
+                raise DecodingError(
+                    f"Expected '{len(indexed_data)}' log topics.  Got '{len(log_topic_types)}'."
+                )
+
+            decoded_topic_data = [
+                decode_single(topic_type, topic_data)
+                for topic_type, topic_data in zip(log_topic_types, indexed_data)
+            ]
+            decoded_log_data = decode_abi(log_data_types, log_data)
+            event_args = dict(
+                itertools.chain(
+                    zip(log_topic_names, decoded_topic_data),
+                    zip(log_data_names, decoded_log_data),
+                )
+            )
+            yield ContractLog(name=abi.name, data=event_args)  # type: ignore
+
+
+def _normalize_abi_types(abi_inputs: List[EventABIType]) -> Iterator[Dict]:
+    for abi_input in abi_inputs:
+        if re.match(r"^{lib_name}r'\.{enum_name}$", str(abi_input.type)):
+            yield {k: "uint8" if k == "type" else v for k, v in abi_input.dict().items()}
+        else:
+            yield abi_input.dict()
 
-        return ContractLog(  # type: ignore
-            name=abi.name,
-            inputs={i.name: event_data[i.name] for i in abi.inputs},  # type: ignore
-        )
+
+def _get_abi_types(abi_inputs: List[Dict]) -> Iterator[Union[str, Dict]]:
+    for abi_input in abi_inputs:
+        abi_type = grammar.parse(abi_input["type"])
+        if abi_type.is_dynamic:
+            yield "bytes32"
```

Comment:
I think using the hash is only for events, so I would make sure this function is only used to decode logs by renaming it